tekton/release.sh

```diff
@@ -0,0 +1,242 @@
+        read -e -p "Enter the Refresh JWT expire time: (eg. 1d) " REFRESH_JWT_EXPIRES_IN
+
+        kubectl -n ${HUB_NAMESPACE} create secret generic api \
+            --from-literal=GH_CLIENT_ID=${GH_CLIENT_ID} \
+            --from-literal=GH_CLIENT_SECRET=${GH_CLIENT_SECRET} \
+            --from-literal=JWT_SIGNING_KEY=${JWT_SIGNING_KEY} \
+            --from-literal=ACCESS_JWT_EXPIRES_IN=${ACCESS_JWT_EXPIRES_IN} \
+            --from-literal=REFRESH_JWT_EXPIRES_IN=${REFRESH_JWT_EXPIRES_IN}
+        
+        kubectl -n ${HUB_NAMESPACE} label secret api app=api 
+
+        kubectl -n ${HUB_NAMESPACE} create cm ui \
+            --from-literal=GH_CLIENT_ID=${GH_CLIENT_ID} \
+            --from-literal=API_URL="https://api.hub.tekton.dev" 
+        
+        kubectl -n ${HUB_NAMESPACE} label cm ui app=ui
+    echo;         
+}
+
+kubectl -n ${HUB_NAMESPACE} get cm api 2>/dev/null >/dev/null || {
+    echo "Hub Config File:"
+        read -e -p "Enter Raw URL of the hub config file (Default: https://raw.githubusercontent.com/tektoncd/hub/master/config.yaml): " HUB_CONFIG
+
+        if [ -z "$HUB_CONFIG" ]
+        then
+            HUB_CONFIG=https://raw.githubusercontent.com/tektoncd/hub/master/config.yaml
+        fi
+
+        kubectl -n ${HUB_NAMESPACE} create cm api \
+            --from-literal=CONFIG_FILE_URL=${HUB_CONFIG} 
+        
+        kubectl -n ${HUB_NAMESPACE} label cm api app=api
+    echo; 
+}
+
+kubectl create namespace ${HUB_CI_NAMESPACE} 2>/dev/null || true
+
+kubectl -n ${HUB_CI_NAMESPACE} delete secret registry-sec --ignore-not-found
+kubectl -n ${HUB_CI_NAMESPACE} get secret registry-sec 2>/dev/null >/dev/null || {
+    echo; echo "Enter Quay registry credentials to push the images: (quay.io/tekton-hub) "
+        read -e -p "Enter Username: " USERNAME
+        read -e -sp "Enter Password: " PASSWORD
+
+        kubectl -n ${HUB_CI_NAMESPACE} create secret generic registry-sec \
+            --type="kubernetes.io/basic-auth"  \
+            --from-literal=username=${USERNAME} \
+            --from-literal=password=${PASSWORD}
+
+        kubectl -n ${HUB_CI_NAMESPACE} annotate secret registry-sec tekton.dev/docker-0=quay.io
+}
+
+echo; echo 'Creates service account and necessary role to create resources: '
+
+kubectl -n ${HUB_CI_NAMESPACE} delete serviceaccount registry-login --ignore-not-found
+cat <<EOF | kubectl -n ${HUB_CI_NAMESPACE} create -f-
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: registry-login
+secrets:
+  - name: registry-sec
+EOF
+
+kubectl -n ${HUB_NAMESPACE} delete role hub-pipeline --ignore-not-found
+kubectl -n ${HUB_NAMESPACE} delete rolebinding hub-pipeline --ignore-not-found
+kubectl -n ${HUB_NAMESPACE} create role hub-pipeline \
+  --resource=deployment,services,pvc,job \
+  --verb=create,get,list,delete,patch
+kubectl -n ${HUB_NAMESPACE} create rolebinding hub-pipeline \
+  --serviceaccount=${HUB_CI_NAMESPACE}:registry-login \
+  --role=hub-pipeline
```

Comment:
can we have one input as well which would as whether user wants to provide `kubeconfig` or not just in case if someone wants to deploy on another cluster?

Reply:
if the user chooses the option of deploying on another cluster then we can avoid setting up the role bindings

Reply:
Also I just realised that whole `secrets` and `configmaps` for the deployment are being setup on the cluster where the deployment pipeline would be running. Is there a pre assumption that the deployment would be performed in the same cluster where tekton pipelines would be running?

Reply:
for now lets target for a single cluster.. later if there is requirement then we can update